hydromt/io.py

```diff
@@ -615,13 +616,25 @@ def open_vector(
     gdf : geopandas.GeoDataFrame
         Parsed geometry file
     """
-    filtered = False
+
+    def _read(f: pyio.IOBase) -> gpd.GeoDataFrame:
+        bbox_reader = gis_utils.prepare_pyogrio_reader_filters(f, bbox, geom, crs)
+        f.seek(0)
+        return gpd.read_file(f, bbox=bbox_reader, mode=mode, **kwargs)
+
     driver = driver if driver is not None else str(fn).split(".")[-1].lower()
     if driver in ["csv", "parquet", "xls", "xlsx", "xy"]:
         gdf = open_vector_from_table(fn, driver=driver, **kwargs)
     else:
-        gdf = gpd.read_file(fn, bbox=bbox, mask=geom, mode=mode, **kwargs)
-        filtered = predicate == "intersects"
+        # check if pathlike
+        if all(
+            map(lambda method: hasattr(fn, method), ("seek", "close", "read", "write"))
+        ):
```

Comment:
nice one, I like this pattern.

Reply:
Totally did not steal it from StackOverflow